docs/CONTRIBUTIONS.md

```diff
@@ -0,0 +1,191 @@
+
+This will install r5py into your environment but allows any further changes
+without the need of reinstalling new version.
+
+### 5. Making changes and writing tests
+
+*r5py* is serious about testing and strongly encourages contributors to embrace
+[test-driven development (TDD)](http://en.wikipedia.org/wiki/Test-driven_development).
+This development process "relies on the repetition of a very short development cycle:
+first the developer writes an (initially failing) automated test case that defines a desired
+improvement or new function, then produces the minimum amount of code to pass that test."
+So, before actually writing any code, you should write your tests. Often the test can be
+taken from the original GitHub issue. However, it is always worth considering additional
+use cases and writing corresponding tests.
+
+*r5py* uses the [pytest testing system](http://doc.pytest.org/en/latest).
+
+### Writing tests
+
+All tests should go into the `tests` directory. This folder contains many
+current examples of tests, and we suggest looking to these for inspiration.
+
+#### Running the test suite
+
+The tests can then be run directly inside your Git clone (without having to
+install *r5py*) by typing::
+
+    pytest
+
+### 6. Updating the Documentation and User Guide
+
+*r5py* documentation resides in the `docs` folder. Changes to the docs are
+make by modifying the appropriate file within `docs`.
+*r5py* docs use a combination of reStructuredText syntax [which is explained here](http://www.sphinx-doc.org/en/stable/rest.html#rst-primer), 
+Jupyter Notebooks ([read more here](https://docs.jupyter.org/en/latest)),
+and the docstrings follow the [Numpy Docstring standard](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt).
+
+Once you have made your changes, you may try if they render correctly by building the docs using sphinx.
+To do so, you can navigate to the docs folder and type::
+
+    make html
+
+The resulting html pages will be located in docs/build/html. In case of any errors,
+you can try to use make html within a new environment based on the libraries in the requirements.txt in the docs folder.
+
+For minor updates, you can skip whole make html part as reStructuredText syntax is
+usually quite straightforward.
+
+#### Updating User Guide
+
+
+Updating user guide might be slightly more complicated as it
+consists of collection of reStructuredText files and Jupyter notebooks.
+Changes in reStructuredText are straightforward, changes in notebooks should be done using Jupyter. Make sure that all cells have their correct outputs as notebooks
+are not executed by readthedocs.
+
+### 7. Formatting the code
+
+#### Python (PEP8 / black)
+
+*r5py* follows the [PEP8](http://www.python.org/dev/peps/pep-0008) standard
+and uses [Black](https://black.readthedocs.io/en/stable/) to ensure a consistent code format throughout the project.
+
+CI will run `black --check` and fails if there are files which would be
```

Comment:
Currently does not pass black, because some things become slightly less readable. Let’s remember to run it ;)

Reply:
Yep, I think this is the goal at least :)